src/Authorization/Controllers/AccessListAuthorizationController.cs

```diff
@@ -0,0 +1,120 @@
+            var orgNumber = HttpContext.User?.Claims.FirstOrDefault(c => c.Type.Equals(AltinnCoreClaimTypes.OrgNumber));
+
+            if (string.IsNullOrWhiteSpace(orgCode?.Value) || string.IsNullOrWhiteSpace(orgNumber?.Value))
+            {
+                return Forbid();
+            }
+
+            ServiceResource resource = await _resourceRegistry.GetResourceAsync(accessListAuthorizationRequest.Resource.Value.ValueSpan.ToString(), cancellationToken);
+            if (resource == null)
+            {
+                return new AltinnValidationProblemDetails([
+                    ValidationErrors.ResourceRegistry_ResourceIdentifier_NotFound.ToValidationError($"/resource/value"),
+                ]).ToActionResult();
+            }
+
+            if (resource.HasCompetentAuthority.Organization != orgNumber.Value)
+            {
+                return new AltinnValidationProblemDetails([
+                    ValidationErrors.ResourceRegistry_CompetentAuthority_NotMatchingAuthenticatedOrganization.ToValidationError($"/HEADER/authorization/bearer/{AltinnCoreClaimTypes.OrgNumber}"),
+                ]).ToActionResult();
+            }
+
+            if (resource.HasCompetentAuthority.Orgcode != orgCode.Value)
+            {
+                return new AltinnValidationProblemDetails([
+                    ValidationErrors.ResourceRegistry_CompetentAuthority_NotMatchingAuthenticatedOrgCode.ToValidationError($"/HEADER/authorization/bearer/{AltinnCoreClaimTypes.Org}"),
+                ]).ToActionResult();
+            }
+        }
+
+        return await AuthorizeInternal(accessListAuthorizationRequest, cancellationToken);
+    }
+
+    /// <summary>
+    /// Internal API for local cluster requests only.
+    /// Authorization of a given subject for resource access through access lists for any service owner organization's access lists and resources. 
+    /// </summary>
+    /// <param name="accessListAuthorizationRequest">Access list authorization request model</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/></param>
+    /// <returns>AccessListAuthorizationResponse</returns>
+    [HttpPost]
+    [Route("authorize/internal")]
+    [ApiExplorerSettings(IgnoreApi = true)]
+    [Consumes(MediaTypeNames.Application.Json)]
+    [Produces(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(typeof(AccessListAuthorizationResponse), StatusCodes.Status200OK)]
+    [ProducesResponseType(typeof(void), StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(typeof(void), StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult> AuthorizeInternal(AccessListAuthorizationRequest accessListAuthorizationRequest, CancellationToken cancellationToken = default)
```

Comment:
Which component will call this API? 
Would it be an idea to use accesstoken for authorization?

Reply:
This is the endpoint meant for internal use from access-management, and yes should add platform token auth here

Reply:
Fixed